native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/meta/ServiceAnalyzer.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.graphql.runtime.engine.meta;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.MapType;
+import io.ballerina.runtime.api.types.MethodType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.RemoteMethodType;
+import io.ballerina.runtime.api.types.ResourceMethodType;
+import io.ballerina.runtime.api.types.ServiceType;
+import io.ballerina.runtime.api.types.StreamType;
+import io.ballerina.runtime.api.types.TableType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.ballerina.stdlib.graphql.runtime.engine.EngineUtils.MUTATION;
+import static io.ballerina.stdlib.graphql.runtime.engine.EngineUtils.QUERY;
+import static io.ballerina.stdlib.graphql.runtime.engine.EngineUtils.SUBSCRIBE_ACCESSOR;
+import static io.ballerina.stdlib.graphql.runtime.engine.EngineUtils.SUBSCRIPTION;
+import static io.ballerina.stdlib.graphql.runtime.utils.ModuleUtils.getModule;
+
+/**
+ * This class analyzes the Ballerina service and extracts the meta information.
+ */
+public class ServiceAnalyzer {
+    private static final BString packageName = StringUtils.fromString(getModule().toString());
+
+    private final Map<String, Resource> resourceMap;
+
+    private static final BString serviceConfigName = StringUtils.fromString("ServiceConfig");
+    private static final BString resourceConfigName = StringUtils.fromString("ResourceConfig");
+    private static final BString queryComplexityConfigKey = StringUtils.fromString("queryComplexityConfig");
+    private static final BString defaultComplexityKey = StringUtils.fromString("defaultFieldComplexity");
+    private static final BString complexityKey = StringUtils.fromString("complexity");
+
+    private final ServiceType serviceType;
+    private final Long defaultQueryComplexity;
```

Comment:
AFAIR when defining fields the preferred way is to define static fields first and then define the instance fields.
```suggestion

    private static final BString serviceConfigName = StringUtils.fromString("ServiceConfig");
    private static final BString resourceConfigName = StringUtils.fromString("ResourceConfig");
    private static final BString queryComplexityConfigKey = StringUtils.fromString("queryComplexityConfig");
    private static final BString defaultComplexityKey = StringUtils.fromString("defaultFieldComplexity");
    private static final BString complexityKey = StringUtils.fromString("complexity");

    private final Map<String, Resource> resourceMap;
    private final ServiceType serviceType;
    private final Long defaultQueryComplexity;
```